packages/eui/src/components/datagrid/body/header/_data_grid_header_row.scss

```
@@ -91,6 +91,8 @@
     .euiPopover-isOpen {
       .euiDataGridHeaderCell__button {
         padding: $euiSizeXS;
+        // balance out additional button target size in header height, prevents increased header cell height
+        margin: -$euiSizeXS 0;
```

minor logical CSS suggestion, avoids setting the horizontal margin when we don't have to 
```suggestion
        margin-block: -$euiSizeXS;
```